modules/libcom/src/osi/osiSock.h

```diff
@@ -28,185 +34,367 @@ struct sockaddr;
+ * that casting between the two more convenient.
+ */
 typedef union osiSockAddr {
     struct sockaddr_in  ia;
     struct sockaddr     sa;
 } osiSockAddr;
 
+/*!
+ * \brief Stores a list of socket addresses
+ *
+ * Container to store a list of socket addresses
+ */
 typedef struct osiSockAddrNode {
     ELLNODE node;
     osiSockAddr addr;
 } osiSockAddrNode;
 
-/*
- * sockAddrAreIdentical()
- * (returns true if addresses are identical)
+/*!
+ * \brief Compares two osiSockAddrs
+ *
+ * Compares two osiSockAddrs
+ *
+ * \return true if addresses are identical, false otherwise.
  */
 LIBCOM_API int epicsStdCall sockAddrAreIdentical 
     ( const osiSockAddr * plhs, const osiSockAddr * prhs );
 
-/*
- *  osiSockDiscoverBroadcastAddresses ()
- *  Returns the broadcast addresses of each network interface found.
- *
- *  This routine is provided with the address of an ELLLIST, a socket,
- *  a destination port number, and a match address. When the
- *  routine returns there will be one additional entry
- *  (an osiSockAddrNode) in the list for each network interface found that
- *  is up and isn't a loop back interface (match addr is INADDR_ANY),
- *  or only the interfaces that match the specified addresses (match addr
- *  is other than INADDR_ANY). If the interface supports broadcasting
- *  then add its broadcast address to the list. If the interface is a
- *  point to point link then add the destination address of the point to
- *  point link to the list.
+/*!
+ * \brief Add available broadcast addresses to a list
+ *
+ *  Add available broadcast addresses to a list. This routine is provided with
+ *  the address of an ELLLIST, a socket, and a match address. When the routine
+ *  returns there will be one additional entry in the list for each network
+ *  interface found that is up and isn't a loop back interface (match addr is
+ *  INADDR_ANY), or only the interfaces that match the specified addresses
+ *  (match addr is other than INADDR_ANY). If the interface supports
+ *  broadcasting then add its broadcast address to the list. If the interface is
+ *  a point to point link then add the destination address of the point to point
+ *  link to the list.
  *
  *  Any mutex locking required to protect pList is applied externally.
  *
+ * Example
+ * =======
+ *  \code{.cpp}
+ *  ELLLIST ifaces = ELLLIST_INIT;
+ *  ELLNODE * cur;
+ *  SOCKET dummy = epicsSocketCreate(AF_INET, SOCK_DGRAM, 0);
+ *  osiSockAddr match;
+ *
+ *  memset(&match, 0, sizeof(match));
+ *  match.ia.sin_family = AF_INET;
+ *  match.ia.sin_addr.s_addr = htonl(INADDR_ANY);
+ *
+ *  osiSockDiscoverBroadcastAddresses(&ifaces, dummy, &match);
+ *  for (cur = ellFirst(&ifaces); cur; cur = ellNext(cur)) {
+ *      char name[64];
+ *      osiSockAddrNode * n = CONTAINER(cur, osiSockAddrNode, node);
+ *      node->addr.ia.sin_port = htons(5555);
+ *      sockAddrToDottedIP(&node->addr.sa, name, sizeof(name);
+ *      fprintf(stderr, "Node is %s", name);
+ *  }
```

Comment:
I might be wrong, but isn't there a need to free the element of the `iface` list afterward? Great example, though!

Reply:
I added these two lines at the end of the sample.

    ellFree(&ifaces);
    epicsSocketDestroy(dummy);